wallet/src/main/scala/org/bitcoins/wallet/Wallet.scala

```diff
@@ -22,31 +23,35 @@ import scodec.bits.BitVector
 
 import scala.concurrent.{ExecutionContext, Future}
 import scala.util.{Failure, Success, Try}
-import org.bitcoins.core.hd.HDCoinType
+import org.bitcoins.core.hd.{
+  HDAccount,
+  HDAddress,
+  HDChainType,
+  HDCoin,
+  HDCoinType,
+  HDPath,
+  HDPurpose,
+  HDPurposes,
+  SegWitHDPath
+}
 import org.bitcoins.core.protocol.blockchain.MainNetChainParams
 import org.bitcoins.core.protocol.blockchain.RegTestNetChainParams
 import org.bitcoins.core.protocol.blockchain.TestNetChainParams
-import org.bitcoins.core.hd.HDChainType
-import org.bitcoins.core.hd.HDAccount
-import org.bitcoins.core.hd.HDAddress
-import org.bitcoins.core.hd.HDCoin
-import org.bitcoins.core.hd.SegWitHDPath
-import org.bitcoins.core.hd.HDPurposes
 
 sealed abstract class Wallet extends UnlockedWalletApi with BitcoinSLogger {
-  implicit def ec: ExecutionContext
 
   val addressDAO: AddressDAO = AddressDAO(dbConfig)
   val mnemonicDAO: MnemonicCodeDAO = MnemonicCodeDAO(dbConfig)
   val accountDAO: AccountDAO = AccountDAO(dbConfig)
   val utxoDAO: UTXOSpendingInfoDAO = UTXOSpendingInfoDAO(dbConfig)
 
-  lazy val hdCoin: HDCoin = {
+  /** The default HD coin */
+  lazy private val DEFAULT_HD_COIN: HDCoin = {
     val coinType = chainParams match {
       case MainNetChainParams                         => HDCoinType.Bitcoin
       case RegTestNetChainParams | TestNetChainParams => HDCoinType.Testnet
     }
-    HDCoin(HDPurposes.SegWit, coinType)
+    HDCoin(Wallet.DEFAULT_HD_PURPOSE, coinType)
```

Comment:
Is there a reason for changing this?

Reply:
You'll notice that it's set to be `HDPurposes.SegWit` in Wallet.scala on line 336. The reason for changing it is making it clear it can be any `HDPurpose`, which at a later point in time can be set by the user either through configuration or by some method call. It's mentioned in the comments above `DEFAULT_HD_PURPOSE`